### DISCUSSION TITLE
Usage from outside?

### DISCUSSION THREAD

**Discussion by Heezer** [Arize-ai/phoenix#1642] [General]
Hi guys and great work first! 😄 
Maybe stupid question, but still relevant for me 😄 Do you support the span JSON files coming from outside? Can you read them and import?
The background is to generate telemetry data from a non Python environment, my case is Java.

Thanks in advance!
Cheers, Alexey

**Comment by mikeldking** [comment]
hey @Heezer ! So in theory yes, phoenix can be launched and mounted on a `TraceDataset` (https://docs.arize.com/phoenix/api/dataset-and-schema#phoenix.tracedataset) - the long term goal of the project will be to have instrumentation and tracers for more runtimes but right now we only have python ones. 😓 

Will try to keep this discussion going as support for different languages emerge and we embark on supporting more than just python. Thanks for your interest in the language!

**Reply by Heezer** [reply]
Hi @mikeldking, thanks for your reply!

The background of my question is - we're looking in our Java AI lib [langchain4j](https://github.com/langchain4j/langchain4j) to support LLM observability and tracing. But I'm not a python guy, so I'm still curious how the integration with Phoenix could happen.

I would appreciate any example here. 🙏 Thanks a lot in advance!!

**Comment by mikeldking** [comment]
@Heezer update for you - we've started moving our instrumentation over to a monorepo that could house Java instrumentation (https://github.com/Arize-ai/openinference) - Phoenix now supports OTEL via OTLP so you can send traces to phoenix using OTEL! Stay tuned.

**Reply by Heezer** [reply]
@mikeldking thanks a lot for update!
